src/Job/NormalJob.ts

```diff
@@ -14,64 +15,110 @@ const DEFAULT_ENV: JobEnv = Object.freeze({
+
+	private environment: JobEnv
 
 	/**
-	 * @param name - The job's name. Must be unique between jobs in the same dependency graph.
-	 * @param prerequisites - A set containing all of this job's prerequisites. Defaults to no prerequisites.
-	 * @param environment - A description of the job's runtime environment. For forwards compatibility, always provide this parameter. Defaults to a Docker image containing GNU Make.
+	 * @param options - the settings for this Job.
+	 *
+	 * The default for options.prerequisiteJobs is the empty set.
+	 *
+	 * The default for options.prerequisiteFiles is the empty set.
+	 *
+	 * The default for options.environment is a Docker image containing GNU Make: {@link DEFAULT_ENV}. For forwards compatibility, always provide this parameter.
 	 */
-	constructor(
-		private readonly name: string,
-		prerequisites: Set<Job> = new Set(),
-		private environment: JobEnv = DEFAULT_ENV,
-	) {
-		// Make a copy so the caller can't directly access prerequisites. Encapsulation!
-		this.prerequisites = new Set(prerequisites)
+	constructor(options: NormalJobOptions) {
+		// Make copies so the caller can't directly access prerequisites. Encapsulation!
+		this.prerequisiteJobs = new Set(options.prerequisiteJobs) // We don't need a deep copy because Jobs are immutable.
+		this.prerequisiteFiles = new Set(options.prerequisiteFiles)
+		this.commands = options.commands.slice()
+		this.target = options.target
+		this.environment = options.environment
+			? {
+					dockerImage: options.environment.dockerImage,
+			  }
+			: DEFAULT_ENV
 	}
 
 	/**
 	 * @returns The job's name.
 	 */
-	public getName(): string {
-		return this.name
+	public getTarget(): string {
+		return this.target
+	}
+
+	/**
+	 * Uses Set's native values() function to get an iterable of prerequisites.
+	 *
+	 * @returns An iterable that iterates over the prerequisiteJobs Set.
+	 */
+	public getPrerequisiteJobsIterable(): Iterable<Job> {
+		return this.prerequisiteJobs.values()
 	}
 
 	/**
 	 * Uses Set's native values() function to get an iterable of prerequisites.
 	 *
-	 * @returns An iterable that iterates over the prerequisites Set.
+	 * @returns An iterable that iterates over the prerequisiteFiles Set.
 	 */
-	public getPrerequisitesIterable(): Iterable<Job> {
-		return this.prerequisites.values()
+	public getPrerequisiteFilesIterable(): Iterable<string> {
+		return this.prerequisiteFiles.values()
 	}
 
 	/**
-	 * Returns the number of prerequisites using Set's native size property.
+	 * Returns the number of prerequisite Jobs using Set's native size property.
 	 */
-	public getNumPrerequisites(): number {
-		return this.prerequisites.size
+	public getNumPrerequisiteJobs(): number {
+		return this.prerequisiteJobs.size
 	}
 
 	/**
 	 * JavaScript calls this function whenever Job is casted to a string.
 	 *
-	 * @returns This job's name.
+	 * @returns A string detailing this job's target, as well as any prerequisite jobs or files.
 	 */
 	public toString(): string {
```

Comment:
Whoever has to debug code using `Job` is going to be really happy to see such detailed toString output 😀